docs/source/queries/timeouts.md

```diff
@@ -0,0 +1,48 @@
+# Query timeouts
+
+Query execution time can be limited by setting a request timeout. If a query does not complete
+in the given time, then `QueryError::RequestTimeout` is returned by the driver immediately,
+so that application logic can continue operating, but the query may still be in progress on the server.
+
+As a side note, if one wishes custom server-side timeouts (i.e. actual interruption of query processing),
+one can use a[`USING TIMEOUT <duration>` directive supported in ScyllaDB](https://github.com/scylladb/scylladb/blob/master/docs/cql/cql-extensions.md#using-timeout)
+(but not in Cassandra).
+
+Timeout can be set globally (per session) or locally (for given statement).
+The default per-session timeout is currently 30s.
+It is possible to turn off timeouts completely by providing `None` as timeout when building `Session`.
+However, setting per-statement timeout to `None` results in falling back to per-session timeout.
+
+```rust
+# extern crate scylla;
+use scylla::{Session, SessionBuilder, query::Query};
+use std::error::Error;
+use std::time::Duration;
+
+async fn timeouts() -> Result<(), Box<dyn Error>> {
```

Comment:
I would personally hide the `use std::error::Error` and `async fn timeouts()` using `#` as well.

The less code there is for the user to read the easier it is for them to understand what's going on.
Like here: https://github.com/scylladb/scylla-rust-driver/blob/5a56e852763ee04e5883180b020810880aea0f93/docs/source/queries/simple.md?plain=1#L32

The lines starting with `#` will be hidden and the user only reads a neat code snippet.

Hiding these lines might be broken at the moment, but hopefully will be fixed soon with #537.